okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -163,29 +200,90 @@ class HttpLoggingInterceptor
 
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
-      val level = this.level
+      var request = chain.request()
+
+      val level = this.requestLevel(request)
 
-      val request = chain.request()
       if (level == Level.NONE) {
         return chain.proceed(request)
       }
 
-      val logBody = level == Level.BODY
-      val logHeaders = logBody || level == Level.HEADERS
+      val logHeaders = level == Level.BODY || level == Level.HEADERS
 
-      val requestBody = request.body
+      val decompressionAlgorithms = findCompressionAlgorithms(chain)
```

Comment:
three features in one PR? configurable levels per-request, streaming, and compression?

I’ve been very reluctant to do a full-featured logging interceptor because I think it’s the wrong tool for most jobs. Using Charles or Chucker or Wireshark all seem better suited, because they don’t have to dump everything to the console. Plus they can naturally display stuff like protos and images